File: C++/two-sum-iii-data-structure-design.cpp
class TwoSum {
private:
    std::unordered_map<int, int> hashmap;
public:
	void add(int number) {
	    if(hashmap.find(number)==hashmap.end())
	        hashmap[number] = 0;
	    hashmap[number] += 1;
	}

	bool find(int value) {
	    for(auto a : hashmap) {
	        if(hashmap.find(value-a.first)!=hashmap.end()) {
	            if(hashmap[a.first]>1 || a.first*2!=value)
	                return true;
	        }
	    }
	    return false;
	}
};

File: Python/two-sum-iii-data-structure-design.py
class TwoSum:
    
    # initialize your data structure here
    def __init__(self):
        self.hashmap = {}

    # @return nothing
    def add(self, number):
        if number not in self.hashmap:
            self.hashmap[number] = 0
        self.hashmap[number] += 1
    # @param value, an integer
    # @return a Boolean
    def find(self, value):
        for i in self.hashmap:
            if (value-i) in self.hashmap:
                if self.hashmap[i]>1 or value!=2*i:
                    return True
        return False